.github/workflows/build.yml

```diff
@@ -254,15 +254,15 @@ jobs:
           ./scripts/feeds update -a
           ./scripts/feeds install -a
 
-      - name: Restore ccache cache
-        id: restore-ccache-cache
-        if: inputs.use_ccache_cache == true
-        uses: actions/cache/restore@v3
-        with:
-          path: openwrt/.ccache
-          key: ccache-${{ inputs.ccache_type }}-${{ inputs.target }}/${{ inputs.subtarget }}-${{ hashFiles('openwrt/include/kernel-**') }}
-          restore-keys: |
-            ccache-${{ inputs.ccache_type }}-${{ inputs.target }}/${{ inputs.subtarget }}-
+      # - name: Restore ccache cache
```

Comment:
Instead of commenting you could add a condition to only run if "s3_key" or something is not set. This way other downstream projects could use the CI without any editing.

Reply:
Yes it's a nice idea will do the change!

Reply:
I added some limitation, only upload and pack with openwrt and use github cache only if we don't have a cache-hit from out S3 Cloud Storage. Should give us also some benchmark on the credit consumption. (on a worst case scenario)

Reply:
Nice, thanks.